cli/cli/commands/github/login/login.go

```diff
@@ -46,5 +54,30 @@ func run(_ context.Context, _ *flags.ParsedFlags, _ *args.ParsedArgs) error {
 		return stacktrace.Propagate(err, "An error occurred setting GitHub user: %v", username)
 	}
 	out.PrintOutLn(fmt.Sprintf("Successfully logged in GitHub user: %v", username))
+	logrus.Info("Restarting engine for GitHub auth to take effect...")
+	err = RestartEngineAfterGitHubAuth(ctx)
+	if err != nil {
+		return err
+	}
+	logrus.Infof("Engine restarted successfully")
+	return nil
+}
+
+func RestartEngineAfterGitHubAuth(ctx context.Context) error {
+	engineManager, err := engine_manager.NewEngineManager(ctx)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating an engine manager.")
+	}
+	var engineClientCloseFunc func() error
+	var restartEngineErr error
+	_, engineClientCloseFunc, restartEngineErr = engineManager.RestartEngineIdempotently(ctx, defaults.DefaultEngineLogLevel, defaultEngineVersion, restartEngineOnSameVersionIfAnyRunning, defaults.DefaultEngineEnclavePoolSize, defaults.DefaultEnableDebugMode, defaults.DefaultGitHubAuthTokenOverride)
```

Comment:
we lose some of these other configurations; guess thats okay

Reply:
ack